pkg/pillar/scripts/device-steps.sh

```diff
@@ -164,6 +167,24 @@ if [ ! -d $PERSISTDIR/log ]; then
     mkdir $PERSISTDIR/log
 fi
 
+
+LAST_EVE_VERSION=""
+if [ -f $LAST_IMAGE_VERSION_FILE ];
+then
+     LAST_EVE_VERSION=$(jq '.Version' < $LAST_IMAGE_VERSION_FILE| tr -d '"')
+fi
+
+# Run upgradeconverter
+if [ "$LAST_EVE_VERSION" = "$EVE_VERSION" ];
+then
+     echo "$(date -Ins -u) device-steps: LAST_EVE_VERSION($LAST_EVE_VERSION) = EVE_VERSION($EVE_VERSION)"
+     echo "$(date -Ins -u) device-steps: Not Running upgradeconverter"
+else
+    echo "$(date -Ins -u) device-steps: Starting upgradeconverter"
+    $BINDIR/upgradeconverter -a "convert"
+    echo "$(date -Ins -u) device-steps: upgradeconverter Completed"
```

Comment:
Here you should be updating the last-eve-version file to make the version check self-contained within the script.

Reply:
Not here.. Because upgrade may fail. I generate the last-image-version in baseosmgr - when the upgrade is successful

Reply:
Ok, so we need to talk about the flow: what happens when upgrade converter fails. It looks like the version is not updates (which is good) but it also looks like this boot of EVE will proceed ahead. On the next boot of EVE the upgrade converter will kick-in again -- but now it may be dealing with a really frankensteing state of /persist. We need to at least document expectations of this @kalyan-nidumolu

Reply:
Right now.. if upgrade converter fails - this will result in the new agent starting with empty global config. This is not treated as catastrophic. For the current use case, it is fine.

When this starts handling more heavy duty work, we should have more elaborate policies on what it should be - Potentially fail the upgrade??